community/detectors/confluence_cve_2021_26084/src/main/java/com/google/tsunami/plugins/detectors/confluence/AtlassianConfluencePreAuthOGNLInjectionDetector.java

```diff
@@ -0,0 +1,135 @@
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.tsunami.plugins.detectors.confluence;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.tsunami.common.net.http.HttpRequest.post;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.flogger.GoogleLogger;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.util.Timestamps;
+import com.google.tsunami.common.data.NetworkServiceUtils;
+import com.google.tsunami.common.net.http.HttpClient;
+import com.google.tsunami.common.net.http.HttpHeaders;
+import com.google.tsunami.common.net.http.HttpResponse;
+import com.google.tsunami.common.net.http.HttpStatus;
+import com.google.tsunami.common.time.UtcClock;
+import com.google.tsunami.plugin.PluginType;
+import com.google.tsunami.plugin.VulnDetector;
+import com.google.tsunami.plugin.annotations.PluginInfo;
+import com.google.tsunami.proto.DetectionReport;
+import com.google.tsunami.proto.DetectionReportList;
+import com.google.tsunami.proto.DetectionStatus;
+import com.google.tsunami.proto.NetworkService;
+import com.google.tsunami.proto.Severity;
+import com.google.tsunami.proto.TargetInfo;
+import com.google.tsunami.proto.Vulnerability;
+import com.google.tsunami.proto.VulnerabilityId;
+import java.io.IOException;
+import java.time.Clock;
+import java.time.Instant;
+import javax.inject.Inject;
+
+/**
+ * A {@link VulnDetector} that detects unprotected Atlassian Confluence Pre-Auth OGNL Injection
+ */
+@PluginInfo(
+    type = PluginType.VULN_DETECTION,
+    name = "AtlassianConfluencePreAuthOGNLInjectionDetector",
+    version = "0.1",
+    description = "This detector checks for unprotected  Atlassian Confluence Pre-Auth OGNL "
+        + "Injection.",
+    author = "C4o (syttcasd@gmail.com)",
+    bootstrapModule = AtlassianConfluencePreAuthOGNLInjectionDetectorBootstrapModule.class)
+public final class AtlassianConfluencePreAuthOGNLInjectionDetector implements VulnDetector {
+
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private static final int number = 10086;
+  private static final String payload = String.format("queryString=Google\\u0027%%2b"
+      + "#{%d*%d}%%2b\\u0027Tsunami", number, number);
+  private static final String pattern = String.format("Google{%d=null}Tsunami", number * number);
```

Comment:
Code style: please use `CONSTANT_CASE` for constants (https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names).